pkg/provisioner/ironic/clients/features.go

```diff
@@ -0,0 +1,57 @@
+package clients
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/utils"
+)
+
+// AvailableFeatures represents features that Ironic API provides.
+// See https://docs.openstack.org/ironic/latest/contributor/webapi-version-history.html
+type AvailableFeatures struct {
+	MaxVersion int
+}
+
+// The minimum microversion that BMO can work with. We need to be really
+// conservative when updating this value since doing so unconditionally breaks
+// operators of older Ironic, even if the feature we need is optional.
+// Update docs/configuration.md when updating the version.
+// Version 1.81 allows retrival of Node inventory
+const baseline = "1.81"
+
+func GetAvailableFeatures(client *gophercloud.ServiceClient) (features AvailableFeatures, err error) {
+	mvs, err := utils.GetSupportedMicroversions(client)
+	if err != nil {
+		return
+	}
+
+	if mvs.MaxMajor != 1 || mvs.MaxMinor < 81 {
+		err = fmt.Errorf("Ironic API 1.81 or newer is required, got %d.%d", mvs.MaxMajor, mvs.MaxMinor)
+		return
+	}
+
+	features.MaxVersion = mvs.MaxMinor
+	return
+}
+
+func (af AvailableFeatures) Log(logger logr.Logger) {
+	// NOTE(dtantsur): update this every time we have more features of interest
+	logger.Info("supported Ironic API features",
+		"maxVersion", fmt.Sprintf("1.%d", af.MaxVersion),
+		"chosenVersion", af.ChooseMicroversion(),
+		"firmwareUpgrades", af.HasFirmwareUpgrades())
```

Comment:
I was thinking that the log messages would read like .. "supported Ironic API features ... firmwareUpgrades = true". WDYT?

Reply:
I think this could me miss interpreted (they would think they can only use to Upgrade firmware, if they had to downgrade maybe they would do manually before using BMO? Updates make more sense I think..)

Reply:
Feel free to adjust the namings in your patch to make them consistent with what you propose.

Reply:
Awesome! Tks @dtantsur